Source/Core/Core/PowerPC/Jit64Common/EmuCodeBlock.cpp

```diff
@@ -971,37 +954,55 @@ void EmuCodeBlock::SetFPRF(Gen::X64Reg xmm)
   else
   {
     MOVQ_xmm(R(RSCRATCH), xmm);
-    TEST(64, R(RSCRATCH), MConst(psDoubleExp));
+    if (single)
+      TEST(32, R(RSCRATCH), MConst(psFloatExp));
```

Comment:
We can use `Imm32` for the constant rather than loading it from the pool in the `single` case.

Reply:
Thanks! I'm not all that used to x64 yet :)

This is only possible for the non-SSE4.1 path, right?

Reply:
No worries, you're doing just fine. 👍

Yeah, SSE instructions don't take full 32-bit immediates, so loading arbitrary constants from memory is the way the go.